src/Reflection/BetterReflection/SourceLocator/FileReadTrapStreamWrapper.php

```diff
@@ -173,4 +181,64 @@ public function url_stat($path, $flags)
 		return $result;
 	}
 
+	/**
+	 * Simulates behavior of reading from an empty file.
+	 *
+	 * @return bool
+	 */
+	public function stream_eof(): bool
+	{
+		return $this->readFromFile;
+	}
+
+	public function stream_flush(): bool
+	{
+		return true;
+	}
+
+	public function stream_tell(): int
+	{
+		return $this->seekPosition;
+	}
+
+	/**
+	 * @param   int  $offset
+	 * @param   int  $whence
+	 * @return  bool
+	 */
+	public function stream_seek($offset, $whence): bool
+	{
+		switch ($whence) {
+			// Behavior is the same for a zero-length file
+			case SEEK_SET:
+			case SEEK_END:
+				if ($offset < 0) {
+					return false;
+				}
+				$this->seekPosition = $offset;
```

Comment:
Should it be possible to set the offset to any positive integer value? An empty file is empty so this seems weird. This note applies to the next case as well.

Reply:
Sorry for the delayed response on this; I tried to characterize the behavior of the various file stream functions in PHP with an empty file to inspire my fake implementation.

You'd think it shouldn't be possible to set the cursor beyond the end of the file, and yet...
```
>>> file_put_contents('test.txt', '')
=> 0
>>> $r = fopen('test.txt', 'rb')
=> stream resource #679
>>> fseek($r, 1234)
=> 0
>>> ftell($r)
=> 1234
>>> fread($r, 1024)
=> ""
>>> ftell($r)
=> 1234
>>> feof($r)
=> true
```

Reply:
Awesome, thank you for the research, the implementation is fine then 👍